fix(its): remove unused code

frenzox opened the pull request.

eloylp commented:
@roberts-ivanovs @tilacog  there's a prototype of the CLI on last commits of this PR's. It doesn't fully work yet, as there are some WIP commits that are not compiling. But i think is close enough to start getting your feedback guys and do improvements once im back.

Some of the things that needs to be fixed/improved (personal note):

* Refactor every names from `contract` to `program` to be more idiomatic.
* Fix relative paths problems. Actually, we don't need to pass the paths to the contracts, but this forces the CLI to be executed in the repo. Maybe it's better to pass absolute paths, so the CLI can work standalone once compiled. 
* Maybe write a little readme for the CLI.
* Complete the last bits of the `Init` subcommand for the `gateway` program. Test its already in place, just need to finish the implementation and make that test green.
* In case we want to update a program, we just need to re-deploy it again with the CLI `Deploy` command. As [docs](https://docs.solanalabs.com/cli/examples/deploy-a-program#redeploy-a-solana-program) states, the important thing here is to preserve `<PROGRAM_NAME>-keypair.json` file next to the program `.so` file. We can actually specify the location of such file with the `-k` argument, overriding the default behaviour.

roberts-ivanovs commented on solana/deployer/src/cli/path.rs:
```diff
@@ -0,0 +1,25 @@
+
+pub fn workspace_root_dir() -> anyhow::Result<PathBuf> {
+    let err_msg = "Cannot find parent path.";
+    Ok(std::env::current_exe()?
```
Would this be equivalent to using `CARGO_TARGET_DIR` env variable [[link to docs]](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-reads)? Im just thinking of how fragile this setup is 🤔

roberts-ivanovs commented on solana/deployer/src/cli.rs:
```diff
@@ -0,0 +1,445 @@
+#[derive(Subcommand)]
+enum Subcommands {
+    Deploy {
+        contracts: Vec<String>,
```
I think lets keep it simple and deploy a single Solana program at a time

roberts-ivanovs commented on solana/deployer/src/cli.rs:
```diff
@@ -0,0 +1,445 @@
+    Deploy {
+        contracts: Vec<String>,
+        #[arg(short, long)]
+        keypair_path: Option<String>,
```
Shouldn't this be mandatory?

roberts-ivanovs commented on solana/deployer/src/cli.rs:
```diff
@@ -0,0 +1,445 @@
+                );
+                info!("Deployed {}", contract);
+            }
+            Ok(Report::Deploy(deployed_contracts))
```
Imo generating a json/toml artifact that we could check in the repo would be a great touch. This will streamline the propagation of "the source of truth" of where our contracts are deployed, their program ids, and the associated PDAs post-initialization (this means that the `Subcommands::Init` step should modify this). But lets discuss this in a separate issue. Lets keep the scope smaller here.

roberts-ivanovs commented on solana/deployer/src/cli.rs:
```diff
@@ -0,0 +1,445 @@
+    fn parse_program_id_from_deploy_output() {
+        let expected_output =
+            Pubkey::from_str("4gG8FWzYihgixVfEdgGkMSdRTN9q8cGyDbkVwR72ir1g").unwrap();
+        let cases = vec![
```
we have `rstest` crate as a dependency in the repo, you can use the [`rstest::case`](https://github.com/la10736/rstest?tab=readme-ov-file#parametrize) macro for this

roberts-ivanovs commented on solana/deployer/src/cli.rs:
```diff
@@ -0,0 +1,445 @@
+
+    use super::*;
+
+    #[test]
```
Because you use `tracing` in the main code, it would be great if we could see the logs during tests as well. You can use the `test_log` crate for that. And might as well use the tokio::test for spinning up a runtime as well, I don't see a reason why we shouldn't do it. [Here's an example of all of the macros being used](https://github.com/eigerco/solana-axelar/blob/1249561eb1da88b1e89d8c6cb085db6054f5d0fd/solana/helpers/evm-contracts-test-suite/src/deployments.rs#L84-L93).

frenzox committed:
chore(its): remove unused token_manager functions

Signed-off-by: Guilherme Felipe da Silva <gfsilva.eng@gmail.com>

frenzox committed:
chore(its): add TODO for FlowLimitSet event

We were recommended to remove this event as it's not being used. Turns
out it not being used is a mistake, we actually have to emit it.
Unfortunately we can't right now though since the current implementation
doesn't support events with Option<T>. Since we're replacing the
mechanism to emit events, it's not worth adding support for Option<T> in
the current implementation, thus adding a TODO just so we don't forget.

Signed-off-by: Guilherme Felipe da Silva <gfsilva.eng@gmail.com>

frenzox committed:
chore(its): remove unused PDA derivation functions

Signed-off-by: Guilherme Felipe da Silva <gfsilva.eng@gmail.com>

frenzox committed:
chore(its): remove Pack from DeployApproval

Signed-off-by: Guilherme Felipe da Silva <gfsilva.eng@gmail.com>

eloylp commented on programs/axelar-solana-its/src/processor/token_manager.rs:
```diff
@@ -54,6 +54,16 @@ pub(crate) fn set_flow_limit<'a>(
+    // }
+    // .emit();
+
     Ok(())
```
Is this something that deserves an issue in our issue tracker ? Just wanted to be sure we do not just forget about it

frenzox replied:
Not sure tbh. Perhaps adding a comment to https://github.com/eigerco/axelar-amplifier-solana/pull/247 should be enough

frenzox replied:
Created https://github.com/eigerco/axelar-amplifier-solana/issues/258 just to be safe

eloylp commented:
Just realised, some checks are failing :(

frenzox committed:
chore(its): remove unused dependency

Signed-off-by: Guilherme Felipe da Silva <gfsilva.eng@gmail.com>
